pkg/reconciler/broker/trigger/trigger.go

```diff
@@ -224,30 +243,57 @@ func (r *Reconciler) subscribeToBrokerChannel(ctx context.Context, b *eventingv1
 				Path:   path.Generate(t),
 			},
 		}
-	}
 
-	if featureFlags.IsOIDCAuthentication() {
-		dest.Audience = pointer.String(FilterAudience)
-	}
+		reply = &duckv1.Destination{
+			URI: &apis.URL{
+				Scheme: "http",
+				Host:   network.GetServiceHostname("broker-filter", system.Namespace()),
+				Path:   path.GenerateReply(t),
+			},
+		}
 
-	// Note that we have to hard code the brokerGKV stuff as sometimes typemeta is not
-	// filled in. So instead of b.TypeMeta.Kind and b.TypeMeta.APIVersion, we have to
-	// do it this way.
-	brokerObjRef := &corev1.ObjectReference{
-		Kind:       brokerGVK.Kind,
-		APIVersion: brokerGVK.GroupVersion().String(),
-		Name:       b.Name,
-		Namespace:  b.Namespace,
+		dls = &duckv1.Destination{
+			URI: &apis.URL{
+				Scheme: "http",
+				Host:   network.GetServiceHostname("broker-filter", system.Namespace()),
+				Path:   path.GenerateDLS(t),
+			},
+		}
 	}
 
-	delivery := t.Spec.Delivery
+	delivery := t.Spec.Delivery.DeepCopy()
 	if delivery == nil {
-		delivery = b.Spec.Delivery
+		delivery = b.Spec.Delivery.DeepCopy()
```

Comment:
Can we add a comment on the rational here?

Something like

// Safety: copy objects to avoid in-place updates bugs

Reply:
Good point. Updated